### PR TITLE
check devel-updates from AUR with last commit

### PR DESCRIPTION
**Pull request by jlaunay**
In order to keep waybar-updates AUR Helper-agnostic I'm now checking if devel packages need an update using the latest commit on the git repository.
This version works for me at the moment, I'm still testing it.
Depending on the number of "devel" packages installed, the first run may take more or less time, but inside "check_updates offline", you only check the packages that have been previously flagged with changes upstream.

### CODE REVIEW

**Comment by jlaunay** [comment]
I think this PR is ready to be reviewed / merged !

[Development packages have a suffix](https://wiki.archlinux.org/title/VCS_package_guidelines#Guidelines) denoting their Version Control System.
Because these packages only download the latest available source when you execute makepkg, their pkgver() in the AUR does not reflect upstream changes.

AUR Helpers (like paru, yay etc.) maintain a database during installation/update to keep track of the last commit, and thus detect when an update is required after a new commit.

Advantage: speed because a database is used.
Disadvantage: if you install a package using an AUR Helper, others won't be able to detect that an update is needed. 

You can enable development package checking (-git) using `waybar-updates -d` (or `--devel`, disabled by default).
The first run may take some time (depending on how many development packages you have) because we need to retrieve the git repository url then use git ls-remote to find the latest commit.

Once done, since the necessary informations are known, we just need to check the last commit.

**Comment by L11R** [comment]
@jlaunay could you run `shellcheck` on it and fix any issues it will highlight? (JetBrains IDEs support it mostly out of the box). Also I am confused by "from Devel" terminology. Devel is not a source like pacman or AUR repos, it's just a kind of AUR-packages. So maybe we will use something like "1 devel-update available from AUR"?

**Comment by L11R** [comment]
I have added ShellCheck GitHub action, but seems like there is no way to launch this check on already created PR...

**Comment by jlaunay** [comment]
> @jlaunay could you run `shellcheck` on it and fix any issues it will highlight? (JetBrains IDEs support it mostly out of the box). Also I am confused by "from Devel" terminology. Devel is not a source like pacman or AUR repos, it's just a kind of AUR-packages. So maybe we will use something like "1 devel-update available from AUR"?

I run this against shellcheck and change for "devel-update available from AUR".
French translation is OK, you should check for "ru" and "tr_TR"

**Comment by L11R** [comment]
@ouztheone do you have time to check Turkey tranlation?

File: po/tr_TR.po [hunk 1]
```diff
@@ -12,6 +12,11 @@ msgstr ""
 "Plural-Forms: nplurals=2; plural=(n != 1);\n"
 "X-Generator: Poedit 3.3.1\n"
 
+msgid "%d devel-update available from AUR"
+msgid_plural "%d devel-updates available from AUR"
+msgstr[0] "AUR'dan %d geliştirme-güncelleme mevcut"
```
Comment (ouztheone) [inline po/tr_TR.po:17]:
```suggestion
msgstr[0] "AUR'dan %d devel-güncellemesi mevcut"
```

File: po/tr_TR.po [hunk 2]
```diff
@@ -12,6 +12,11 @@ msgstr ""
...
+msgstr[1] "AUR'dan %d geliştirme-güncelleme mevcut"
```
Comment (ouztheone) [inline po/tr_TR.po:18]:
```suggestion
msgstr[1] "AUR'dan %d devel-güncellemesi mevcut"
```

**Review by ouztheone — CHANGES_REQUESTED** [review]

**Comment by ouztheone** [comment]
> @ouztheone do you have time to check Turkey translation?

Hey, thanks for the heads up. I added my suggestions. :)

**Review by ouztheone — APPROVED** [review]

**Comment by L11R** [comment]
@jlaunay I will release update tomorrow, out of desk now!

**Comment by jlaunay** [comment]
> @jlaunay I will release update tomorrow, out of desk now!

Thanks a lot, in the meantime, I published a [waybar-updates-git](https://aur.archlinux.org/packages/waybar-updates-git) package on AUR.

**Pull request merged**